### DISCUSSION TITLE
How to set the output sample rate

### DISCUSSION THREAD

**Discussion by lijuncode** [sbooth/SFBAudioEngine#236] [Q&A]
The SFBAudioPlayer output sample rate is lower than the file sample rate, how to set it?

**Comment by sbooth** [comment]
In your instance of `SFBAudioPlayer`, call the function `withEngine(block:)` and set the bus format on the connection(s) between `playerNode` and `engine.mainMixerNode` to the desired sample rate.

**Reply by lijuncode** [reply]
how to set the bus format

**Comment by sbooth** [comment]
When calling `connect` on `AVAudioEngine` you can pass the bus format: [connect(_:to:format:)](https://developer.apple.com/documentation/avfaudio/avaudioengine/1388974-connect)

Using this method you can build the processing graph with a format you specify.
